pkg/provisioner/ironic/ironic.go

```diff
@@ -1346,3 +1368,110 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+func (p *ironicProvisioner) buildAndValidateCleanSteps(hwConfigs []interface{}) (cleanSteps []nodes.CleanStep) {
+	for _, config := range hwConfigs {
```

Comment:
This method should be able to look at p.host for itself, and should not need values passed in as argument. That eliminates the need for the loop, the type assertion, and the switch statement. For now there is always at most 1 thing to do (RAID). Later, when we add BIOS setting support, this same method can add the appropriate steps.

Reply:
Done. I get the RAID config directly from `p.host.Spec` and remove the for loop, type assertion, and switch statement :)